### PR TITLE
Usage example with stateless fastify-secure-session

### PR DESCRIPTION
**Pull request by mashpie**
Life-saving project. Thank you so much!

Even though I was able to get started within minutes, I wanted to share another example: How to use grant with `fastify-secure-session` instead of `fastify-session`.

`fastify-secure-session` doesn't require nor provides a server-side store as it's designed to work stateless, which is super helpful in distributed systems. The caveat compared to most default implementations is that you can't modify `req.session` properties directly but have to use `req.session.get()` and `req.session.set()` methods.

Being close to submit a PR for a modification of your handler in `lib/handler/fastify.js` to support getters/setters around lines 32ff:

```js
      var {location, session, state} = await grant({
        method: req.method,
        params: req.params,
        query: qs.parse(req.query),
        body: qs.parse(req.body),
        state: req.grant,
        session: typeof req.session.get === 'function'? req.session.get('grant'):req.session.grant,
      })

      if(typeof req.session.set === 'function'){
        req.session.set('grant', session)
      }else{
        req.session.grant = session
      }
```

...well, in general, I try to act as least obstructive as possible, so I decided to provide another option without the need for code-change, as fastify provides us with hooks and encapsulation to design or extend most aspects of any given module/plugin. 

So in short: Applied another hook to define `req.session.grant` as getter/setter was enough to get me going:

```js
  fastify.addHook('onRequest', (req, res, next) => {
    Object.defineProperty(req.session, 'grant', {
      get: () => req.session.get('grant'),
      set: (value) => req.session.set('grant', value)
    })
    next()
  })
```

Happy to share these thoughts and an example :)

### CODE REVIEW

**Comment by simov** [comment]
Thanks, this is really helpful :+1: 

I appreciate the effort you put into making it work outside of Grant. I might consider integrating your patch as well.
